crates/bevy_asset/src/loader.rs

```diff
@@ -89,9 +89,10 @@ impl<'a> LoadContext<'a> {
         self.labeled_assets.insert(None, asset);
     }
 
-    pub fn set_labeled_asset(&mut self, label: &str, asset: LoadedAsset) {
+    pub fn set_labeled_asset<T: Asset>(&mut self, label: &str, asset: LoadedAsset) -> Handle<T> {
```

Comment:
This change makes this kind of code fail compilation:

```rust
load_context.set_labeled_asset("cube", LoadedAsset::new(mesh));
```

Is there a way to not be bothered with specifying the `T` parameter?

Reply:
Hmm yeah that is suboptimal. Its both less ergonomic and creates a class of bug where the user specifies the wrong asset type.

We could make it `LoadedAsset<T>`, then created a BoxedLoadedAsset that we coerce it into within `set_labeled_asset`